Patch-Source: https://gitlab.gnome.org/GNOME/tinysparql/-/merge_requests/767
---
From 4fe4e9ddd40fcec1e9fd307c0c61e25287394df2 Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 21 Mar 2025 10:22:37 +0100
Subject: [PATCH] tests: Set up main loop earlier in startup data struct

This will ensure the main thread will receive the main loop
after initialization is completed.

Closes: https://gitlab.gnome.org/GNOME/tinysparql/-/issues/482
---
 tests/libtinysparql/tracker-deserialize-test.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tests/libtinysparql/tracker-deserialize-test.c b/tests/libtinysparql/tracker-deserialize-test.c
index 45a1f9094..2716002da 100644
--- a/tests/libtinysparql/tracker-deserialize-test.c
+++ b/tests/libtinysparql/tracker-deserialize-test.c
@@ -361,7 +361,7 @@ thread_func (gpointer user_data)
 	context = g_main_context_new ();
 	g_main_context_push_thread_default (context);
 
-	main_loop = g_main_loop_new (context, FALSE);
+	main_loop = data->thread_loop = g_main_loop_new (context, FALSE);
 
 	endpoint = tracker_endpoint_dbus_new (data->direct, data->dbus_conn, NULL, NULL, NULL);
 
@@ -373,7 +373,6 @@ thread_func (gpointer user_data)
 	if (!endpoint)
 		return NULL;
 
-	data->thread_loop = main_loop;
 	g_main_loop_run (main_loop);
 
 	g_main_loop_unref (main_loop);
-- 
GitLab

